tests/integration_tests/tests/agentless_tests/component/test_component_basics.py

```diff
@@ -76,6 +76,150 @@ def test_component_creation_with_secrets_and_plugins(self):
+  component_node:
+    type: cloudify.nodes.Component
+    properties:
+      resource_config:
+        blueprint:
+          external_resource: true
+          id: basic
+        deployment:
+          id: component
+      secrets:
+        integer: 1
+        list: [1, 2]
+        dict:
+            a: 1
+        string: a
+        float: 1.5
+        boolean: false
+        regex: ^.$
+"""
+        blueprint_path = self.make_yaml_file(main_blueprint)
+        self.deploy_application(blueprint_path, deployment_id=deployment_id)
+        self.assertEquals(self.client.secrets.get('integer')['value'], '1')
+        self.assertEquals(self.client.secrets.get('list')['value'], '[1, 2]')
+        self.assertEquals(self.client.secrets.get('dict')['value'], '{"a": 1}')
+        self.assertEquals(self.client.secrets.get('string')['value'], 'a')
+        self.assertEquals(self.client.secrets.get('float')['value'], '1.5')
+        self.assertEquals(self.client.secrets.get('boolean')['value'], 'False')
+        self.assertEquals(self.client.secrets.get('regex')['value'], '^.$')
+
+
+class ComponentInputsTypesTest(AgentlessTestCase):
+    def test_basic_types(self):
+        component_blueprint = """
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - cloudify/types/types.yaml
+
+inputs:
+    integer:
+        type: integer
+    list:
+        type: list
+    dict:
+        type: dict
+    string:
+        type: string
+    float:
+        type: float
+    boolean:
+        type: boolean
+    regex:
+        type: regex
+"""
+        blueprint_path = self.make_yaml_file(component_blueprint)
+        self.client.blueprints.upload(blueprint_path,
+                                      entity_id='basic')
+        deployment_id = 'd{0}'.format(uuid.uuid4())
+        main_blueprint = """
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - cloudify/types/types.yaml
+
+node_templates:
+  component_node:
+    type: cloudify.nodes.Component
+    properties:
+      resource_config:
+        blueprint:
+          external_resource: true
+          id: basic
+        deployment:
+          id: component
+          inputs:
+            integer: 1
+            list: [1, 2]
+            dict:
+                a: 1
+            string: a
+            float: 1.5
+            boolean: false
+            regex: ^.$
+"""
+        blueprint_path = self.make_yaml_file(main_blueprint)
+        self.deploy_application(blueprint_path, deployment_id=deployment_id)
+        deployment = self.client.deployments.get('component')
+        self.assertEquals(deployment.inputs['integer'], 1)
+        self.assertEquals(deployment.inputs['list'], [1, 2])
+        self.assertEquals(deployment.inputs['dict'], {'a': 1})
+        self.assertEquals(deployment.inputs['string'], 'a')
+        self.assertEquals(deployment.inputs['float'], 1.5)
+        self.assertEquals(deployment.inputs['boolean'], False)
+        self.assertEquals(deployment.inputs['regex'], '^.$')
```

Comment:
The assertions are the same, maybe use a method to do them?

Reply:
because it's a one liner it will not really change something